pkg/3scale/amp/manual-templates/amp/build.yml

```diff
@@ -11,12 +11,12 @@ objects:
     annotations:
     labels:
       app: zync
-    name: ruby-24-centos7
+    name: ruby-25-ubi7
   spec:
     tags:
     - from:
         kind: DockerImage
-        name: centos/ruby-24-centos7
+        name: registry.access.redhat.com/ubi7/ruby-25
```

Comment:
From https://access.redhat.com/RegistryAuthentication
```
Over the coming year, Red Hat will standardize on registry.redhat.io and registry.connect.redhat.com as the primary container registries for Red Hat and certified partners, and will eventually decommission registry.access.redhat.com. While customers can continue to use registry.access.redhat.com until it is decommissioned, it is recommended that you prepare to use registry.redhat.io.
```

Using the new `registry.redhat.io` registry requires authenticated access. The circleCI job building images should create secret with the operator developer account credentials living in circleCi. Like this:

```
  create-redhat-registry-io-secret:
    steps:
      - run:
          name: Create registry.redhat.io secret
          command: |
            oc create secret docker-registry threescale-registry-auth \
              --docker-password="${REGISTRY_REDHAT_IO_PASSWORD}" \
              --docker-username="${REGISTRY_REDHAT_IO_USERNAME}" \
              --docker-server="${REGISTRY_REDHAT_IO_SERVER}"
```

Reply:
I think that's not exactly like that @eguzki.

UBI images are a special case where they keep being published in registry.access.redhat.com if I'm not mistaken.

Reply:
https://www.redhat.com/en/blog/introducing-red-hat-universal-base-image
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/building_running_and_managing_containers/index#using_init_red_hat_base_images

Reply:
Should be
`registry.redhat.io/ubi7/ruby-25`

https://catalog.redhat.com/software/containers/detail/5c889d63bed8bd28d0e79ee2?container-tabs=gti&gti-tabs=registry-tokens

Reply:
Ok, @eguzki. I'll make sure to include that in the workflow and change the base domain of the image. Thanks.

Reply:
For example, operator-sdk itself uses registry.access:
https://github.com/operator-framework/operator-sdk/blob/424a61d56000e6e3d91d352faa1bd4f7c814661f/internal/scaffold/build_dockerfile.go

It seems with the specific case of ubi images it's sort of a "grey" area

Reply:
`registry.access.redhat.com` is deprecated as a registry. Will eventually be decommissioned. If some RedHat blog uses that registry is their fault.

Reply:
It's not a blog only, it's the official documentation in rhel8 and some of the products

Reply:
Then it is not deprecated? If it is not, I am ok with using it.

Reply:
https://access.redhat.com/articles/4238681 another article where it says the authenticated or unauthenticated registry can be used.

Regarding whether it's deprecated or not... in general registry.access.redhat.com is deprecated. However, in the special case of ubi images I'm not so sure about that. UBI are meant to be easily accessible. Using an authenticated registry to get them kind of defeats the purpose and I guess that's why lots of documentation and product are using registry.access.redhat.com. I don't have any official confirmation about being one way or the other though. If you are interested we can try to ask someone that is involved in that

If we were to use registry.redhat.io wouldn't forks experience problems with it as they don't have access to the secret defined in CircleCI?

Reply:
Also, is some external team using the file? is QE using it? If we decide to change to .io we have to tell them too

Reply:
Some additional information I've found: https://bugzilla.redhat.com/show_bug.cgi?id=1719994#c4

Reply:
@miguelsorianod there is no issue about using the new repository `registry.redhat.io/ubi7/ruby-25`.

The job building the nightly images is not run by GH forks and external PR's. 

Even if we needed to run in all PR's (including external PR's), we could use some robot developer account. `registry.redhat.io` is authenticated, but free. Anyone can get developer account credentials. Thus, it is not a big deal exposing some test robot developer account red hat developer credentials in our circleci job.

Reply:
As I said, based on the information I've seen and shown you, I don't see the need to move to registry.redhat.io at the moment in the special case of UBI images. It seems it's going to stay there for the foreseeable future and it adds a layer of authentication to the usage of that template that it's not necessary at all, increasing the needed steps/complexity to use it.

> Even if we needed to run in all PR's (including external PR's), we could use some robot developer account. registry.redhat.io is authenticated, but free. Anyone can get developer account credentials. Thus, it is not a big deal exposing some test robot developer account red hat developer credentials in our circleci job.

I'm not a fan of placing credentials in the repository, even if "anyone can get developer account credentials".

I prefer not doing it at the moment for the reasons I already exposed. If the situation were to change in the future regarding UBI images this could be changed of course.

That said, being limited to the nightly jobs only, and that those are not run on all PRs, it's not a blocking change for me. If you decide to do it cc @3scale/qe has to be aware of this. Also ideally in that case in zync @guicassolato it would be better to be consistent with the decision taken here.

Reply:
> in zync @guicassolato it would be better to be consistent with the decision taken here.

I think this is particularly relevant to https://github.com/3scale/zync/pull/330.

Reply:
+1 on no need to move to registry.redhat.io yet. 

Let's only do that when $whoever_responsible deems it absolutely necessary, because using the repo that **requires** auth adds an extra step. 

Yes, it's free, and, yes everyone can get an account, but it doesn't mean that they will. :) 
(they might simply give up)

If we _can_ use a registry that doesn't require auth, let's save everyone the additional hassle.

Reply:
Guys, RH decided to move to authenticated registry. Not a big deal. Get used to it.

Reply:
I am not sold on that, tbh. (Otherwise, why is the unauthenticated registry around 2+ years later and docs all over still mention the unauthenticated one, as per @miguelsorianod's comments?)